timestampvm/src/api/chain_handlers.rs

```diff
@@ -60,17 +60,26 @@ pub struct GetBlockResponse {
 }
 
 /// Implements API services for the chain-specific handlers.
-pub struct Service {
-    pub vm: vm::Vm,
+pub struct Service<A>
+where
+    A: AppSender + Send + Sync + Clone + 'static,
+{
+    pub vm: vm::Vm<A>,
 }
 
-impl Service {
-    pub fn new(vm: vm::Vm) -> Self {
+impl<A> Service<A>
+where
+    A: AppSender + Send + Sync + Clone + 'static,
+{
+    pub fn new(vm: vm::Vm<A>) -> Self {
```

Comment:
Any reason we don't want to remove the type-stuttering (`vm::Vm`)? I find this hurts readability, especially for tutorials.

Reply:
Yeah I mean its fair to ask we should probably fold up everything into ChainVM to be honest.

Reply:
improved